components/script/dom/htmliframeelement.rs

```diff
@@ -131,30 +140,70 @@ impl HTMLIFrameElement {
 
         let global_scope = window.upcast::<GlobalScope>();
         let load_info = IFrameLoadInfo {
-            load_data: load_data,
             parent_pipeline_id: global_scope.pipeline_id(),
             frame_id: self.frame_id,
-            old_pipeline_id: old_pipeline_id,
             new_pipeline_id: new_pipeline_id,
-            sandbox: sandboxed,
             is_private: private_iframe,
             frame_type: frame_type,
             replace: replace,
         };
-        global_scope
-              .constellation_chan()
-              .send(ConstellationMsg::ScriptLoadedURLInIFrame(load_info))
-              .unwrap();
+
+        if load_data.as_ref().map_or(false, |d| d.url.as_str() == "about:blank") {
```

Comment:
Can we not send the frame-creation message to the constellation when the iframe attributes are processed?